node/src/blockchain/test_utils.rs

```diff
@@ -223,106 +355,23 @@ impl BlockchainInterfaceMock {
-
-    fn prepare(&self, method: &str, params: Vec<rpc::Value>) -> (RequestId, rpc::Call) {
-        let request = web3::helpers::build_request(1, method, params.clone());
-        let mut prepare_params = self.prepare_params.lock().unwrap();
-        prepare_params.push((method.to_string(), params));
-        (prepare_params.len(), request)
-    }
-
-    fn send(&self, id: RequestId, request: rpc::Call) -> Self::Out {
-        self.send_params.lock().unwrap().push((id, request.clone()));
-        match self.send_results.borrow_mut().pop_front() {
-            Some(response) => Box::new(futures::finished(response)),
-            None => {
-                println!("Unexpected request (id: {:?}): {:?}", id, request);
-                Box::new(futures::failed(Web3Error::Unreachable))
-            }
-        }
-    }
-}
-
-impl BatchTransport for TestTransport {
-    type Batch = web3::Result<Vec<Result<rpc::Value, web3::Error>>>;
-
-    fn send_batch<T>(&self, requests: T) -> Self::Batch
-    where
-        T: IntoIterator<Item = (RequestId, rpc::Call)>,
-    {
-        self.send_batch_params
-            .lock()
-            .unwrap()
-            .push(requests.into_iter().collect());
-        let response = self.send_batch_results.borrow_mut().remove(0);
-        Box::new(futures::finished(response))
-    }
-}
-
-impl TestTransport {
-    pub fn prepare_params(mut self, params: &Arc<Mutex<Vec<(String, Vec<rpc::Value>)>>>) -> Self {
-        self.prepare_params = params.clone();
-        self
-    }
-
-    //why prepare_result missing? Look up for a comment at the struct
-
-    pub fn send_params(mut self, params: &Arc<Mutex<Vec<(RequestId, rpc::Call)>>>) -> Self {
-        self.send_params = params.clone();
-        self
-    }
-
-    pub fn send_result(self, rpc_call_response: rpc::Value) -> Self {
-        self.send_results.borrow_mut().push_back(rpc_call_response);
-        self
-    }
-
-    pub fn send_batch_params(
-        mut self,
-        params: &Arc<Mutex<Vec<Vec<(RequestId, rpc::Call)>>>>,
-    ) -> Self {
-        self.send_batch_params = params.clone();
-        self
-    }
-
-    pub fn send_batch_result(
-        self,
-        batched_responses: Vec<Result<rpc::Value, web3::Error>>,
-    ) -> Self {
-        self.send_batch_results.borrow_mut().push(batched_responses);
-        self
-    }
-
-    pub fn initiate_reference_counter(mut self, reference_arc: &Arc<()>) -> Self {
-        self.reference_counter_opt = Some(reference_arc.clone());
-        self
-    }
-}
-
 pub fn make_fake_event_loop_handle() -> EventLoopHandle {
     Http::with_max_parallel("http://86.75.30.9", REQUESTS_IN_PARALLEL)
         .unwrap()
         .0
 }
 
+pub fn make_default_signed_transaction() -> SignedTransaction {
+    SignedTransaction {
+        message_hash: Default::default(),
+        v: 0,
+        r: Default::default(),
+        s: Default::default(),
+        raw_transaction: Default::default(),
+        transaction_hash: Default::default(),
```

Comment:
Can you just check out if the web3 library's SignedTransaction structure by any chance implements the Default trait as a whole. It would make this much briefer... eh... it would be a straight substitution for this entire piece of code. That's why I believe you had thought of this yourselves.

But this kind of check shouldn't be hard, so please do it to be sure.